### PR TITLE
"ZeroDivisionError: integer division or modulo by zero" while backpropagating #51653

### PR DESCRIPTION
**Pull request by sannamir**
"ZeroDivisionError: integer division or modulo by zero" while backpropagating #51653
logic reference:
https://github.com/tensorflow/tensorflow/issues/51653#issuecomment-904756234

### CODE REVIEW

**Comment by gbaned** [comment]
@sannamir  Can you please sign CLA. Thanks!

**Comment by sannamir** [comment]
@googlebot I signed it!

File: tensorflow/python/ops/array_grad.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 from __future__ import absolute_import
 from __future__ import division
 from __future__ import print_function
-
+import numpy
```
Comment (allenlavoie) [inline tensorflow/python/ops/array_grad.py:20]: This seems unused

File: tensorflow/python/ops/array_grad.py [hunk 2]
```diff
@@ -115,6 +115,8 @@ def _ExtractInputShapes(inputs):
     if context.executing_eagerly() or isinstance(concat_dim, ops.EagerTensor):
       # Using mod here for convenience since concat_dim is already verified
       # in concat implementation to be within the allowed [-rank, rank) range.
+      if(input_value[0].shape==(1,)):
```
Comment (allenlavoie) [inline tensorflow/python/ops/array_grad.py:118]: Wouldn't this cause problems if it shape was (), not (1,)? Not sure I see what this is fixing; consider adding an explanatory comment.

**Review by allenlavoie — COMMENTED** [review]
Please add a test for the behavior you're fixing.

**Comment by gbaned** [comment]
@sannamir  Can you please check @allenlavoie's comments and resolve conflicts?. Thanks!

**Comment by gbaned** [comment]
@sannamir Any update on this PR? Please. Thanks!

**Comment by sannamir** [comment]
It has passed all checks.

**Comment by mihaimaruseac** [comment]
It has a conflict

**Comment by gbaned** [comment]
@sannamir  Can you please resolve conflicts? Thanks!

**Comment by gbaned** [comment]
@sannamir Any update on this PR? Please. Thanks!

**Comment by gbaned** [comment]
I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen!

**Pull request closed without merging**